### DISCUSSION TITLE
gem5 generated as shared library

### DISCUSSION THREAD

**Discussion by JohnVida** [gem5/gem5#1083] [Q&A]
I am currently working on a project that involves using gem5 for co-simulation purposes with VCS, and I aim to compile gem5 as a shared library. Given my limited familiarity with scons, I am reaching out to seek your expertise in this area.

Could you possibly provide guidance or an example on how to properly configure scons to generate a shared library from gem5? Any specific flags or configurations that need to be set would be incredibly helpful.

Thank you in advance for your time and assistance. I look forward to your expert advice.

**Comment by powerjg** [comment]
Hello, welcome to the gem5 commnunity!

You can build gem5 as a shared object out-of-the-box.

For example, if you want to use the ALL defaults, you can use the following.

```
scons build/ALL/gem5.opt
scons --with-cxx-config --without-python build/ALL/libgem5_opt.so
```

The flag `--with-cxx-config` says that you will not be executing the configuration in python but using a predefined `config.ini` file.
The flag `--without-python` tells gem5 to build without python support. I don't think this is required, but may be useful for you.

Since you're running gem5 without python, you may find the code in `util/cxx_config` useful.

The README may be helpful as well: https://github.com/gem5/gem5/blob/stable/util/cxx_config/README
